### ISSUE TITLE
"Unable to load .NET runtime, specifically hostfxr." when launching.

### ISSUE THREAD

**Issue by KRC2000** [godotengine/godot#75186]
### Godot version

4.0.1

### System information

 Linux Mint 21.1 Vera base: Ubuntu 22.04 jammy

### Issue description

When launching Godot 4.0.1 .NET an error screen is shown:
![image](https://user-images.githubusercontent.com/47914319/226622824-97acf62d-6e26-4fac-9778-ff917f9db58e.png)
dotnet is installed, included in the path:
![image](https://user-images.githubusercontent.com/47914319/226623714-f5e9a7be-64b8-4783-8f72-1cfc5e51752e.png)
![image](https://user-images.githubusercontent.com/47914319/226623946-30de93e4-c413-4b02-a2e3-9691a5946b30.png)

### Steps to reproduce

Download https://github.com/godotengine/godot/releases/download/4.0.1-stable/Godot_v4.0.1-stable_mono_linux_x86_64.zip
Run on the Linux system with installed dotnet

### Minimal reproduction project

None

**Comment by KRC2000** [comment]
Just tried on my Pop_OS installation on the same machine with dotnet installed  from the apt - works as intended.
![image](https://user-images.githubusercontent.com/47914319/226654331-1986add9-9b06-4cd0-a784-c3c3247094a6.png)

**Comment by RedworkDE** [comment]
Interestingly only the blog posts for pre-releases actually document this, but dotnet needs to be installed in a standard location for the install to be detected. For linux that means that the dotnet executable must be found at the path `/usr/share/dotnet`. IIRC it should be possible to symlink your install from that path, but I don't recall what exactly has to be linked where for things to actually work correctly.

**Comment by mrombout** [comment]
I ran into the same issue even when installing .NET SDK 6.0 (`dotnet-sdk-6.0`) from [Microsoft's package repository](https://learn.microsoft.com/en-us/dotnet/core/install/linux-ubuntu#register-the-microsoft-package-repository) into the standard location `/usr/share/dotnet/` on Ubuntu 22.04.

Installing .NET SDK 7.0 (`dotnet-sdk-7.0`) instead solved the issue for me. Perhaps it's some kind of compatibility issue?

**Comment by Noxiefilioxie** [comment]
it was doing the same for me
But i used the Debian 11 guide here
https://learn.microsoft.com/en-us/dotnet/core/install/linux-debian

which helped for me

**Comment by yingshaoxo** [comment]
Try this:

```bash
wget https://dot.net/v1/dotnet-install.sh -O dotnet-install.sh
chmod 777 dotnet-install.sh 
./dotnet-install.sh --channel 7.0

export PATH="$PATH:$HOME/.dotnet"
```

> Use microsoft script to install dotnet. It works fine.

**Comment by retropele** [comment]
> Try this:
> 
> ```shell
> wget https://dot.net/v1/dotnet-install.sh -O dotnet-install.sh
> chmod 777 dotnet-install.sh 
> ./dotnet-install.sh --channel 7.0
> 
> export PATH="$PATH:$HOME/.dotnet"
> ```
> 
> > Use microsoft script to install dotnet. It works fine.

I installed everything according to Microsoft's instructions, got the error in the original post here. This was the only thing that got it to actually work for me - thanks :)

**Comment by LeeWannacott** [comment]
Following this fixed it for me: https://stackoverflow.com/questions/73753672/a-fatal-error-occurred-the-folder-usr-share-dotnet-host-fxr-does-not-exist

**Comment by Noahnoah55** [comment]
> Try this:
> 
> ```shell
> wget https://dot.net/v1/dotnet-install.sh -O dotnet-install.sh
> chmod 777 dotnet-install.sh 
> ./dotnet-install.sh --channel 7.0
> 
> export PATH="$PATH:$HOME/.dotnet"
> ```
> 
> > Use microsoft script to install dotnet. It works fine.

Hey sorry if this bumps the issue but I found while this did get rid of the fatal error that crashes Godot on loading a file, I was still getting a Red error saying

```ERROR: The host fxr folder does not exist: /usr/share/dotnet/host/fxr```

![error](https://github.com/godotengine/godot/assets/17057873/bb3a17a4-9cf1-448a-8393-6cfd9f74cac6)

To fix this, you just have to set the DOTNET_ROOT environment variable so that Godot can find the fxr file.

After updating your path to include dotnet, you can simply run the following to automatically find the right directory and point DOTNET_ROOT to it.

```export DOTNET_ROOT="$(dirname $(which dotnet))"```

**Comment by runevision** [comment]
I'm getting this issue too on macOS (M1 Mac). I installed .NET runtime without changing any paths or any other settings, just clicking "Next" until finished. So whatever location Godot expects is not the one where it's installed by default.

**Comment by fchamieh** [comment]
I had same issue, my dotnet was installed in my home directory. Just simlinked /usr/share/dotnet -> /home/fadi/.dotnet  and Goddot.NET works fine now

**Comment by salciunas** [comment]
I had this problem also, this article helped me: https://bobbyhadz.com/blog/fatal-error-ocurred-the-folder-usr-share-dotnet-host-fx-does-not-exist

**Comment by sjkillen** [comment]
> > Try this:
> > ```
> > wget https://dot.net/v1/dotnet-install.sh -O dotnet-install.sh
> > chmod 777 dotnet-install.sh 
> > ./dotnet-install.sh --channel 7.0
> > 
> > export PATH="$PATH:$HOME/.dotnet"
> > ```
> > 
> > 
> > > Use microsoft script to install dotnet. It works fine.
> 
> Hey sorry if this bumps the issue but I found while this did get rid of the fatal error that crashes Godot on loading a file, I was still getting a Red error saying
> 
> `ERROR: The host fxr folder does not exist: /usr/share/dotnet/host/fxr`
> 
> ![error](https://user-images.githubusercontent.com/17057873/268075787-bb3a17a4-9cf1-448a-8393-6cfd9f74cac6.png)
> 
> To fix this, you just have to set the DOTNET_ROOT environment variable so that Godot can find the fxr file.
> 
> After updating your path to include dotnet, you can simply run the following to automatically find the right directory and point DOTNET_ROOT to it.
> 
> `export DOTNET_ROOT="$(dirname $(which dotnet))"`

Also ran into this issue, but I fixed it with
`sudo ln -sT ~/.dotnet /usr/share/dotnet`
Note that if you installed .NET through the package manager as well there will already be files here and this won't work

**Comment by Schweeeeeeeeeeeeeeee** [comment]
Same on arch even after running these commands:
```

wget https://dot.net/v1/dotnet-install.sh -O dotnet-install.sh

chmod +x ./dotnet-install.sh

./dotnet-install.sh --version latest

./dotnet-install.sh --version latest --runtime aspnetcore

./dotnet-install.sh --channel 8.0


```
From here https://learn.microsoft.com/en-us/dotnet/core/install/linux-scripted-manual#scripted-install

**Comment by raulsntos** [comment]
@Schweeeeeeeeeeeeeeee Scripted installs are not recommended for setting up a development environment, their intended [purpose](https://learn.microsoft.com/en-us/dotnet/core/tools/dotnet-install-script#purpose) is to use it for CI scenarios. The recommended installation method is to use your distro's package manager.

**Comment by Schweeeeeeeeeeeeeeee** [comment]
@raulsntos Thx that worked :) ill keep that in mind

**Comment by patoblando** [comment]
> Try this:
> 
> ```shell
> wget https://dot.net/v1/dotnet-install.sh -O dotnet-install.sh
> chmod 777 dotnet-install.sh 
> ./dotnet-install.sh --channel 7.0
> 
> export PATH="$PATH:$HOME/.dotnet"
> ```
> 
> > Use microsoft script to install dotnet. It works fine.

This fixed it for my, trying to install Godot_v4.2.1 on Kubuntu 23.10 x86_6

**Comment by Rikitik** [comment]
Same Issue on windows CMD "dotnet --info" works just fine for me. 
Godot Version 4.2.2
dotnet 6.0.413

**Comment by kongshanpai01** [comment]
add an environment variable named "DOTNET_ROOT", setting the value with your path of dotnet like "E:\Program Files\.net SDK" or "/home/user_name/dotnet"(depends on your OS platform), to let Godot recongnize your customize dotnet path

**Comment by dbehmoaras** [comment]
MacOS m3 Sequoia 15.1 (my on the go tool).

I followed the fix suggested [here](https://github.com/godotengine/godot/issues/73747#issuecomment-1451292178) which suggests using .NET 6.0, but I'm using .NET 8 in my dev environment outside of Godot (e.g. C# scripts in VSCode etc...). I'm concerned about compatibility - has anyone figured out how to get it running with .NET 8 (or 9 even) ? Managing 2 different versions of .NET could become unwieldy.

**Comment by TenCoKaciStromy** [comment]
The issue is still there. I tried `Godot_v4.6-stable_mono_linux.x86_64` (Ubuntu 25.10).

It works if I run the godot from terminal. But if I run it from Files, it shows this error.

The dotent (version 10) was installed through scripted install by recommendation of Microsoft. The installation is on path `~/.dotnet`.

**Comment by TenCoKaciStromy** [comment]
You are right, I expresed myself incorrectly. I meant "the way Microsoft describe", so not "script built by myself".

After reainstal through apt, it works.

Thanks for help!
